API/scanner/scanner-adapter-openapi-v1.0.yaml

```diff
@@ -0,0 +1,392 @@
+openapi: 3.0.0
+info:
+  title: Harbor Scanner Adapter API
+  description: |
+    ## Overview
+
+    The API that has to be implemented in order to plug an artifact scanner in [Harbor](https://goharbor.io) registry.
+
+    The [/scan](#operation/AcceptScanRequest) and [/scan/{scan_request_id}/report](#operation/GetScanReport) operations
+    do the actual scanning and provide a scan report which can be displayed in the Harbor web console.
+
+    The [/scan](#operation/AcceptScanRequest) operation is asynchronous. It should just enque the scan job for
+    processing a scan request and return the identifer which allows polling a corresponding scan report with the
+    [/scan/{scan_request_id}/report](#operation/GetScanReport) operation. Harbor will call the
+    [/scan/{scan_request_id}/report](#operation/GetScanReport) operation periodically until it returns 200 or 500 status
+    codes.
+
```

Comment:
The [/scan](#operation/AcceptScanRequest) operation is asynchronous. It should enqueue the job for processing a scan request and return the identifier. This allows the ???? operation to poll a corresponding scan report with the [/scan/{scan_request_id}/report](#operation/GetScanReport) operation. Harbor will call the [/scan/{scan_request_id}/report](#operation/GetScanReport) operation periodically until it returns 200 or 500 status codes.

**Question**: which operation does the polling?

Reply:
the job @stuclem

Reply:
or Harbor. Both are ok @stuclem

Reply:
Thanks @steven-zou. In that case: "This allows Harbor to poll a corresponding scan report with the /scan/{scan_request_id}/report operation."